### DISCUSSION TITLE
[Feat Req] Elasticsearch: Support for metric queries without group by

### DISCUSSION THREAD

**Discussion by omar-m-othman** [grafana/grafana#61167] [Data Sources]
Grouping is not always necessary. It took me like two hours to figure out that was my only problem.

Consider this example:
```
Query => field1:term1 AND field2:term2
Metric => Unique Count(field3)
```

That's all what I wanted! Now, seeing that the results didn't make any sense, I opened the Network tab and saw the extra layer of aggregation coming from the obligatory "Group By". I just removed it and queried ElasticSearch myself, which produced the correct number!

Am I missing something too obvious?!

**Comment by torkelo** [comment]
You can remove the group by if you want (if your using table panel, graph panel requires date histogram)

**Comment by torkelo** [comment]
Sorry, forgot that you need a group by. You can change the interval to 1y or some big number to only get one value back

**Comment by omar-m-othman** [comment]
LoL. This is not a "resolution" for the issue. I did what you mentioned as a hack, and then I opened the ticket here. A GitHub ticket is for discussing the problem and reaching a proper resolution, not for hacking the problem around. The latter can be done on IRC!

**Comment by torkelo** [comment]
Sorry, yes your right it's not an ideal solution. It's just so common that an ES query requires a group by that it has never come up as a feature request before.

**Comment by time-change** [comment]
Is there any follow-up for this? I have a simple use case where I just want to graph the value of one specific field for each document.

**Comment by jakubvojacek** [comment]
+1

**Comment by SchroterQuentin** [comment]
+1

**Comment by scroodj** [comment]
+1

**Comment by Mugglephilip** [comment]
+1

**Comment by fatelgit** [comment]
+1

**Comment by matyax** [comment]
We need to do some research here.

**Comment by jdlugole** [comment]
+1

**Comment by jazerda** [comment]
+1

**Comment by wchaws** [comment]
+1

**Comment by ivanahuckova** [comment]
For additional discussion, please use https://github.com/grafana/grafana/issues/77557

**Reply by torbenaa** [reply]
What ?? That has nothing to do with the original request ??

**Comment by torbenaa** [comment]
+1 more

**Comment by douglasg14b** [comment]
Well, this explains why I can't get any single values queries to work for opensearch.....

**Comment by tohar2d2** [comment]
How is disabling group by in a query still not added after 9+ years? Is there another another way that doesn't involve adding totally unnecessary compute overhead? Or is hacking around this with 1 big bucket still the only solution?

**Comment by matyax** [comment]
Tracking in https://github.com/grafana/grafana-elasticsearch-datasource/issues/412 . Please redirect any further conversation to the feature request. Thank you.
